bin/easystar-0.4.3.js

```diff
@@ -285,17 +308,6 @@ var EasyStar =
 	    *
 	    **/
 	    this.findPath = function (startX, startY, endX, endY, callback) {
-	        // Wraps the callback for sync vs async logic
-	        var callbackWrapper = function (result) {
-	            if (syncEnabled) {
-	                callback(result);
-	            } else {
-	                setTimeout(function () {
-	                    callback(result);
-	                });
-	            }
-	        };
-
 	        // No acceptable tiles were set
```

Comment:
This is likely the most controversial change ☝️ , I'm happy to discuss this, besides from deferring the callback onto the next execution stack, I'm not sure what is gained from wrapping this in a `setTimeout` for the asynchronous mode.

Reply:
This is important as without it we would be introducing zalgo bugs. The fact that the callback is asynchronous should be consistent.

See here: http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony

Reply:
That is a really interesting read, not something that I've ever come across. I'm certain I 100% understand it in our context, but I'm more than happy to revert this.

Reply:
In our context it is relevant because

> If you have an API which takes a callback,
and sometimes that callback is called immediately,
and other times that callback is called at some point in the future,
then you will render any code using this API impossible to reason about, and cause the release of Zalgo.

If sync is not enabled, we want to consistently call the callback asynchronously. Removing the `setTimeout` means the callback _could_ be called immediately, making the callback behavior inconsistent.